pulpcore/cli/common/generic.py

```diff
@@ -331,55 +331,53 @@ def _version_callback(
 
 
-# TODO: would be great to have enable this to take a validator, rather than having
-# to build "on top of" it like I'm doing now w/ json_callback
-def load_file_or_string_callback(
-    ctx: click.Context, param: click.Parameter, value: Optional[str]
-) -> Any:
-    """Load string from input or from file if string starts with @."""
-    the_content: str
+def load_file_wrapper(handler: Callable[[click.Context, click.Parameter, str], Any]) -> Any:
+    """A wrapper that used for chaining or decorating callbacks that manipulate with input data."""
 
-    # pass None and "" verbatim
-    if not value:
-        return value
+    @wraps(handler)
+    def _load_file_or_string_wrapper(
+        ctx: click.Context, param: click.Parameter, value: Optional[str]
+    ) -> Any:
+        """Load the string from input, or from file if the value starts with @."""
+        if not value:
+            return value
 
-    if value.startswith("@"):
-        the_file = value[1:]
-        try:
-            with click.open_file(the_file, "r") as fp:
-                the_content = fp.read()
-        except OSError:
-            raise click.ClickException(
-                _("Failed to load content from {file}").format(file=the_file)
-            )
-    else:
-        the_content = value
+        if value.startswith("@"):
+            the_file = value[1:]
+            try:
+                with click.open_file(the_file, "r") as fp:
+                    the_content = fp.read()
+            except OSError:
+                raise click.ClickException(
+                    _("Failed to load content from {file}").format(file=the_file)
+                )
+        else:
+            the_content = value
 
-    return the_content
+        return handler(ctx, param, the_content)
 
+    return _load_file_or_string_wrapper
 
-def load_json_callback(ctx: click.Context, param: click.Parameter, value: Optional[str]) -> Any:
-    """Load JSON from input string or from file if string starts with @."""
 
-    # None or empty-str are legal - shortcircuit here
-    if not value:
-        return value
+load_string_callback = load_file_wrapper(lambda c, p, x: x)
+
+
+def json_callback(ctx: click.Context, param: click.Parameter, value: str) -> Any:
+    if value is None:
+        return None
 
-    # Now try to evaluate legal JSON
-    json_object: Any
-    json_string: str = load_file_or_string_callback(ctx, param, value)
     try:
-        json_object = json.loads(json_string)
+        json_object = json.loads(value)
     except json.decoder.JSONDecodeError:
         raise click.ClickException(_("Failed to decode JSON"))
     else:
         return json_object
 
 
-def labels_callback(
-    ctx: click.Context, param: click.Parameter, value: Optional[str]
-) -> Optional[Dict[str, str]]:
-    # None is legal - shortcircuit here
+load_json_callback = load_file_wrapper(json_callback)
+
+
+def load_labels_callback(ctx: click.Context, param: click.Parameter, value: str) -> Any:
```

Comment:
We can be more specific here.
```suggestion
def load_labels_callback(ctx: click.Context, param: click.Parameter, value: str) -> Optional[Dict[str, str]]:
```
(Don't forget to run black...)
Also when you change the name, can you verify that this is not used anywhere in the known plugins?

This should be my last request. We are ready to go then...

Reply:
I run `git grep` against pulp-cli, pulp-cli-deb, and pulp-cli-ostree. Only these occurrences were found:
```
pulpcore/cli/common/generic.py:def load_labels_callback(
pulpcore/cli/common/generic.py:    callback=load_labels_callback,
```

I think we are good to go.